test/e2e/tests/test_adopt_endpoint.py

```diff
@@ -0,0 +1,243 @@
+
+from acktest.aws import s3
+from acktest.resources import random_suffix_name
+from acktest.k8s import resource as k8s
+
+from e2e import (
+    service_marker,
+    CRD_GROUP,
+    CRD_VERSION,
+    ENDPOINT_CONFIG_RESOURCE_PLURAL,
+    MODEL_RESOURCE_PLURAL,
+    ENDPOINT_RESOURCE_PLURAL,
+    create_adopted_resource,
+    wait_sagemaker_endpoint_status,
+    assert_endpoint_status_in_sync,
+    ENDPOINT_STATUS_INSERVICE,
+)
+from e2e.replacement_values import REPLACEMENT_VALUES
+
+
+@pytest.fixture(scope="module")
+def name_suffix():
+    return random_suffix_name("sdk-endpoint", 32)
+
+
+@pytest.fixture(scope="module")
+def sdk_endpoint(name_suffix, sagemaker_client):
+    model_name = name_suffix + "-model"
+    endpoint_config_name = name_suffix + "-config"
+    endpoint_name = name_suffix
+    data_bucket = REPLACEMENT_VALUES["SAGEMAKER_DATA_BUCKET"]
+    model_input = {
+        "ModelName": model_name,
+        "Containers": [
+            {
+                "Image": REPLACEMENT_VALUES["XGBOOST_IMAGE_URI"],
+                "ModelDataUrl": f"s3://{data_bucket}/sagemaker/model/xgboost-mnist-model.tar.gz",
+            }
+        ],
+        "ExecutionRoleArn": REPLACEMENT_VALUES["SAGEMAKER_EXECUTION_ROLE_ARN"],
+    }
+    endpoint_config_input = {
+        "EndpointConfigName": endpoint_config_name,
+        "ProductionVariants": [
+            {
+                "VariantName": "variant-1",
+                "ModelName": model_name,
+                "InitialInstanceCount": 1,
+                "InstanceType": "ml.c5.large",
+            }
+        ],
+    }
+    endpoint_input = {
+        "EndpointName": endpoint_name,
+        "EndpointConfigName": endpoint_config_name,
+    }
+
+    model_response = sagemaker_client.create_model(**model_input)
+    assert model_response.get("ModelArn", None) is not None
+    endpoint_config_response = sagemaker_client.create_endpoint_config(
+        **endpoint_config_input
+    )
+    assert endpoint_config_response.get("EndpointConfigArn", None) is not None
+    endpoint_response = sagemaker_client.create_endpoint(**endpoint_input)
+    assert endpoint_response.get("EndpointArn", None) is not None
+
+    yield (
+        model_input,
+        model_response,
+        endpoint_config_input,
+        endpoint_config_response,
+        endpoint_input,
+        endpoint_response,
+    )
+    wait_sagemaker_endpoint_status(
+        sagemaker_client, endpoint_name, ENDPOINT_STATUS_INSERVICE
+    )
+    sagemaker_client.delete_endpoint(EndpointName=endpoint_name)
+    sagemaker_client.delete_endpoint_config(EndpointConfigName=endpoint_config_name)
+    sagemaker_client.delete_model(ModelName=model_name)
+
+
+@pytest.fixture(scope="module")
+def adopted_endpoint(sdk_endpoint):
+    (model_input, _, endpoint_config_input, _, endpoint_input, _) = sdk_endpoint
+
+    replacements = REPLACEMENT_VALUES.copy()
+
+    replacements["ADOPTED_RESOURCE_NAME"] = "adopt-" + model_input["ModelName"]
```

Comment:
Same comment. Looks like you can pull a common method out of these configuration specifications.

Reply:
Else it atleast requires some comments to make the method easier to read.

Reply:
will add comments here